src/components/Tooltip/Tooltip.test.tsx

```diff
@@ -0,0 +1,233 @@
+
+    expect(screen.getByText(label)).not.toBeVisible()
+  })
+  it("can be dismissed by clicking anchor element", async () => {
+    // Arrange
+    const children = "Hello"
+    const label = "World"
+
+    // Act
+    render(
+      <Tooltip hideOnClick label={label}>
+        {children}
+      </Tooltip>,
+    )
+
+    const receivedChildren = screen.getByText(children)
+
+    // Assert
+    await userEvent.hover(receivedChildren)
+
+    expect(screen.getByText(label)).toBeVisible()
+
+    await userEvent.click(receivedChildren)
+
+    expect(screen.getByText(label)).not.toBeVisible()
+  })
+  it("can always be shown", async () => {
+    // Arrange
+    const children = "Hello"
+    const label = "World"
+
+    // Act
+    render(
+      <Tooltip alwaysShow label={label}>
+        {children}
+      </Tooltip>,
+    )
+
+    const receivedChildren = screen.getByText(children)
+
+    // Assert
+    await userEvent.tab()
+
+    await userEvent.keyboard("[Escape]")
+
+    expect(screen.getByText(label)).toBeVisible()
+
+    await userEvent.hover(receivedChildren)
+
+    await userEvent.unhover(receivedChildren)
+
+    expect(screen.getByText(label)).toBeVisible()
+  })
+  /* 
+  We are making this async since this component internally sets state within a ref callback,
+  otherwise testing library complains about state being set and not wrapping fireEvent in 'act' 
+  */
+  it("can display an optional arrow element", async () => {
+    // Arrange
+    const children = "Hello"
+    const label = "World"
+    const arrowTestID = "tooltip_arrow"
+
+    // Act
+    render(
+      <Tooltip showArrow label={label}>
+        {children}
+      </Tooltip>,
+    )
+
+    const received = screen.getByTestId(arrowTestID)
+
+    // Assert
+    await waitFor(() => expect(received).toBeInTheDocument())
+  })
+  it("allows for an enterDelay or exitDelay to be set", async () => {
+    // Arrange
+    jest.useFakeTimers()
+    /*
+      when using fake timers we have to set delay to null, 
+      otherwise the userEvent never resolves
+    */
+    const user = userEvent.setup({ delay: null })
+    const children = "Hello"
+    const label = "World"
+    const enterDelay = 300
+    const exitDelay = 500
+
+    // Act
+    render(
+      <Tooltip enterDelay={enterDelay} exitDelay={exitDelay} label={label}>
+        {children}
+      </Tooltip>,
+    )
+
+    const receivedChildren = screen.getByText(children)
+
+    // Assert
+    await user.hover(receivedChildren)
+
+    expect(screen.getByText(label)).not.toBeVisible()
+
+    act(() => {
+      jest.advanceTimersByTime(enterDelay)
+    })
+
+    expect(screen.getByText(label)).toBeVisible()
+
+    await user.unhover(receivedChildren)
+
+    expect(screen.getByText(label)).toBeVisible()
+
+    act(() => {
+      jest.advanceTimersByTime(exitDelay)
+    })
+
+    expect(screen.getByText(label)).not.toBeVisible()
+  })
+  it("supports aria-describedby", async () => {
+    // Arrange
+    jest.useRealTimers()
```

Comment:
This looks a bit odd, one test is calling `useRealTimers` _probably_ because the previous test is using `useFakeTimers`, there's now an implicit relationship between tests that one test _has_ to come after the other.

Would this be better if the test calling `useFakeTimers` also called `useRealTimers` at the end like some sort of "cleanup"? And then other tests might not have to care whether or not `fakeTimers` are being used.

Reply:
Yeah you are right. It makes sense to contain the timer logic within that particular case. A cleanup will be perfect 😄